libflux/src/ast/walk/tests.rs

```diff
@@ -0,0 +1,306 @@
+// NOTE: These test cases directly match ast/json_test.go.
+// Every test is preceded by the correspondent test case in golang.
+use super::*;
+use crate::parser::parse_string;
+
+fn test_walk(source: &str, want: Vec<&str>) {
+    let file = parse_string("test_walk", source);
+    let mut nodes = Vec::new();
+    walk(
+        &create_visitor(&mut |n| nodes.push(format!("{}", n))),
+        Node::File(&file),
+    );
+    assert_eq!(want, nodes);
+}
+
+#[test]
```

Comment:
I believe unit tests usually go in the same file as the code they are testing.

Reply:
They can be in either, for really long tests its nice to put them in their own file so the files are easier to navigate. These tests are not too long so I could move them back. I was following the convention of the parser and ast modules, which do have very large tests.